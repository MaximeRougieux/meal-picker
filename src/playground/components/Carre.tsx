import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

export const Carre = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Hello</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 300,
    height: 300,
    backgroundColor: 'red',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  text: {
    color: 'green',
    fontSize: 70,
  },
});
