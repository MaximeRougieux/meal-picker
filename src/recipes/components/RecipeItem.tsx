import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

interface RecipeProps {
  name: string;
  imageUrl: string;
  category: string;
}

export const RecipeItem = ({ name, imageUrl, category }: RecipeProps) => {
  return (
    <View style={styles.main}>
      <Image style={styles.image} source={{ uri: imageUrl }} />
      <View style={styles.textContainer}>
        <Text style={styles.title}>{name}</Text>
        <Text style={styles.categoryText}>{category}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  main: {
    flexDirection: 'row',
    alignItems: 'center',
    margin: 8,
    height: 200,
  },
  image: {
    height: '100%',
    aspectRatio: 1,
    marginRight: 8,
  },
  textContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    height: '100%',
  },
  title: {
    fontSize: 24,
  },
  categoryText: {
    fontSize: 12,
  },
});
