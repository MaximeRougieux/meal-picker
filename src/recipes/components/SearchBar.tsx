import React from 'react';
import { StyleSheet, TextInput } from 'react-native';
import debounce from 'lodash/debounce';
interface SearchBarProps {
  setKeyword: (text: string) => void;
}
const SearchBar = ({ setKeyword }: SearchBarProps) => {
  return (
    <TextInput
      onChangeText={debounce(setKeyword, 300)}
      testID="searchbar"
      placeholder="Search"
      style={styles.main}
    />
  );
};

const styles = StyleSheet.create({ main: { fontSize: 18, margin: 6 } });

export default SearchBar;
