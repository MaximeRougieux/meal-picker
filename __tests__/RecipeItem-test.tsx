import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { RecipeItem } from '../src/recipes/components/RecipeItem';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <RecipeItem
        name="Hello"
        imageUrl="https://reactnative.dev/img/tiny_logo.png"
        category="test"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
