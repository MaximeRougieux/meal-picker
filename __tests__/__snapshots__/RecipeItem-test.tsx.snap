// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 200,
      "margin": 8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactnative.dev/img/tiny_logo.png",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "height": "100%",
        "marginRight": 8,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Hello
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
